platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/dsl/DefaultComponentMetadataContext.java

```diff
@@ -24,12 +24,10 @@ class DefaultComponentMetadataContext implements ComponentMetadataContext {
 
     private final ComponentMetadataDetails details;
     // We keep this field for access from Groovy scripts, as we currently miss some public API: https://github.com/gradle/gradle/issues/12349
```

Comment:
❌ This change should be reverted, see the comment above. Suppress it explicitly instead.